### ISSUE TITLE
* Reading/writing to Hard disk images is always failing with cache=none

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vgmurphy/limbo-android#1]
```
What steps will reproduce the problem?
1. Use cache=none
2. Boot a freedos cd image
3. During installation to a hard disk (image) the installation fails while 
reading partitions with a message "Could not read partition".

What is the expected output? What do you see instead?
Block reads/writes should succeed instead the guest OS usually fails 
unsuccessful to read.

Please use labels and text to provide additional information.



```

Original issue reported on code.google.com by `ideal2i...@gmail.com` on 23 Jun 2012 at 7:04

**Comment by GoogleCodeExporter** [comment]
```
 suspect this is due to memory buffer alignments for the host before read/write.
```

Original comment by `ideal2i...@gmail.com` on 23 Jun 2012 at 8:06

**Comment by GoogleCodeExporter** [comment]
```
O_DIRECT and O_SYNC had wrong values.
Fix applied to version  v0.9.1b-1 HD Cache configuration option is disabled 
until issue #3 is fixed. Closing Issue.
```

Original comment by `ideal2i...@gmail.com` on 10 Sep 2012 at 11:08

**Comment by GoogleCodeExporter** [comment]
```
The change for this is done now.
```

Original comment by `ideal2i...@gmail.com` on 10 Sep 2012 at 11:11
- Changed state: **Fixed**
